chaosazure/__init__.py

```diff
@@ -48,15 +64,33 @@ def auth(secrets: Secrets) -> ServicePrincipalCredentials:
         resource_client = ResourceManagementClient(cred, azure_subscription_id)
         compute_client = ComputeManagementClient(cred, azure_subscription_id)
     ```
+
+    Again, if you are not working with Public Azure Cloud,
+    and you set azure_cloud in secret,
+    this will pass one more parameter `base_url` to above function.
+    ```python
+        cloud = eval(secrets.get("azure_cloud"))
+        client = ComputeManagementClient(
+            credentials=cred, subscription_id=subscription_id,
+                        base_url=cloud.endpoints.resource_manager)
+    ```
+
     """
     creds = dict(
-        azure_client_id=None, azure_client_secret=None, azure_tenant_id=None)
+        azure_client_id=None, azure_client_secret=None,
+        azure_tenant_id=None, azure_cloud=None)
 
     if secrets:
         creds["azure_client_id"] = secrets.get("client_id")
         creds["azure_client_secret"] = secrets.get("client_secret")
         creds["azure_tenant_id"] = secrets.get("tenant_id")
 
+        if secrets.get("azure_cloud") is None:
+            azure_cloud = "AZURE_PUBLIC_CLOUD"
+        else:
+            azure_cloud = secrets.get("azure_cloud")
+        creds["azure_cloud"] = azure_cloud
```

Comment:
@xpdable In general I like your pull request a lot. It regards the different Azure cloud environments.

For the marked lines of code I'm just wondering wether it is a good idea to declare the field 'azure_cloud' in the secrets section. I think that the field 'azure_cloud' is not secret at all :). I would rather suggest to declare it in the configuration section. That would make more sense for the user in the end. What do you think?